### PR TITLE
Add support for Firefox

### LINKED ISSUE #38: Add support for Firefox
Apparently, it'd be much appreciated as a Firefox extension as well. The only blocker is our usage of the shadow DOM. Since we're just using it to isolate our CSS from the page's we could easily get rid of it. We'd need to reset the styles for each element but we can probably get it right.

So, to publish it as a Firefox extension we need to:

1. Use a fallback when shadow DOM is not supported
    * [ ] Detect shadow DOM support
    * [ ] Use inline styles instead of `<style />` with CSS classes (maybe there's a webpack loader for that)
    * [ ] Create an exhaustive CSS reset and apply it to every single elements
2. Adapt the current configuration to Firefox

I don't know much about Firefox extensions but since they decided to follow the manifest 2.0, it shouldn't be too hard.

### PR DESCRIPTION
**Pull request by Zhouzi**
Close #38

### CODE REVIEW

**Comment by Zhouzi** [comment]
Note to self:

* [ ] Add warning when visiting the demo page with something else than Chrome or Firefox
* [ ] Change button to install extension depending on browser (Chrome/Firefox)

**Comment by Zhouzi** [comment]
The web extension polyfill is currently broken, it throws:

> Constructor WeakMap requires 'new'

Which is caused by the polyfill extending `WeakMap`. Will probably need to submit a PR to this package.

**Comment by Zhouzi** [comment]
I couldn't get the shadow DOM polyfill to work so I used another trick to isolate our CSS. It should be all good now, I just submitted the extension to Firefox and it's pending review. Once it's published, I'll just add the URL to the extension in the demo page and this PR will be ready for review/merge.

File: manifest.json [hunk 1]
```diff
@@ -11,8 +11,7 @@
     "128": "icon128.png"
   },
   "options_ui": {
-      "page": "options.html",
-      "chrome_style": true
```
Comment (Zhouzi) [inline manifest.json]: Yep, you're right! Apparently, Firefox defaults to true for `browser_style` but I just updated the build script to explicitly declare it.

Related: [documentation](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/options_ui).

**Comment by Zhouzi** [comment]
Just received a comment on the review:

> Please note the following:
> 
> 1) Please refactor your code to make review feasible.
> ref: content.js
> 
> 2) Please refrain from including resources as lengthy data:URIs strings, as doing so has performance implications and considerably complicates the review process and adds to your waiting time. Please include them as separate files within your XPI.

2 should be pretty straightforward. Fixing 1 is a bit more though as it probably means we need to ditch Webpack in order to generate less bloated code and include the files "one by one". We can keep using Babel, as long as its output remains readable.

**Comment by Zhouzi** [comment]
Actually, we can attach the sources and we should be fine, see [How to get a quicker review for add-ons with source code attached](https://blog.mozilla.org/addons/2016/02/24/quicker-reviews-with-sources-attached/).

**Comment by Zhouzi** [comment]
In order to get a clean version of the source code, I'm now cloning the repository into `dist/firefox/sources`. I'll have to update the git clone once this branch is about to be merged (currently cloning this branch).

**Comment by Soreine** [comment]
I find this review process requirement really word

**Comment by Zhouzi** [comment]
My bad, I'm learning as I'm going through the process. We don't actually need to pack the sources with the extension but we can upload them in the dashboard. I just submitted a new version and uploaded the sources, now pending review.

**Comment by Soreine** [comment]
Alright, seems more sensible to provide the source for the review...

**Comment by Soreine** [comment]
Good job! 🎉

**Pull request merged**